docs/error-handling.md

```diff
@@ -0,0 +1,26 @@
+## Error handling
+
+The `pureconfig.error` package provides semantically rich case classes to
+describe failures reading from a `ConfigValue`. When implementing your own
+`ConfigConvert`s, you're recommended to use the provided case classes.
+
+The `ConfigReaderFailure` class has an optional location field that can be used
+to point to the physical location of a `ConfigValue` that raised an error. When
+using the `ConfigReaderFailure` sealed family of case classes, you can use the
+`ConfigValueLocation.apply(cv: ConfigValue)` method to automatically create an
+optional location from a `ConfigValue`.
+
+Given this setup:
+
+```scala
+import com.typesafe.config._
+import pureconfig.error._
+
+val cv = ConfigFactory.load.root().get("conf")
+```
+
+We can try to load a missing key and get a useful error:
+```scala
+KeyNotFound("xpto", ConfigValueLocation(cv))
+// res1: pureconfig.error.KeyNotFound = KeyNotFound(xpto,Some(ConfigValueLocation(/Users/dvm105/pureconfig/docs/target/scala-2.12/classes/application.conf,11)))
```

Comment:
Perhaps what we could do to make this have a stable output here is to just print a substring of the filename and the line number. For example:
```
val notFound =  KeyNotFound("xpto", ConfigValueLocation(cv))
notFound.filename.takeRight(24)
notFound.lineNumber
```